src/rgw/rgw_rados.cc

```diff
@@ -3946,6 +4012,22 @@ int RGWRados::fetch_remote_obj(RGWObjectCtx& obj_ctx,
     if (ret < 0) {
       goto set_err_state;
     }
+
+    if (cct->_conf->rgw_copy_verify_object) {
+      string trimmed_etag = etag;
+
+      /* Remove the leading and trailing double quotes from etag */
+      trimmed_etag.erase(std::remove(trimmed_etag.begin(), trimmed_etag.end(),'\"'),
+        trimmed_etag.end());
+
+      if (cb.get_calculated_etag().compare(trimmed_etag)) {
+        ret = -EIO;
+        ldout(cct, 0) << "ERROR: source and destination objects don't match. Expected etag:"
+          << trimmed_etag << " Computed etag:" << cb.get_calculated_etag() << dendl;
```

Comment:
we need to do this check before the call to `processor.complete()` - that's the transaction that makes the object visible